src/server.ts

```diff
@@ -53,17 +54,36 @@ export class AttendingServer {
             member.roles.cache.find(role => role.name == queue.name) !== undefined)
     }
 
-    async RemoveMemberFromQueues(member: GuildMember): Promise<number> {
+    async RemoveMemberFromQueues(member: GuildMember, interaction_type: string): Promise<number> {
         let queue_count = 0
         await Promise.all(this.queues.map(queue => {
             if (queue.Has(member)) {
                 queue_count++
-                return queue.Remove(member)
+                return queue.Remove(member, interaction_type)
             }
         }))
         return queue_count
     }
 
+    async RemoveMember(queue_name: string, member: GuildMember, interaction_type: string): Promise<void> {
+        const queue = this.queues.find(queue => queue.name == queue_name)
+        if(interaction_type === 'APPLICATION_COMMAND'){
```

Comment:
The conditions here seem redundant. Is checking the interaction type necessary?

Reply:
removed the checking on interaction type since now we're returning UserError regardless of it's a slash command or a button press